src/snarky/parties-leaves-json.ts

```diff
@@ -0,0 +1,78 @@
+import { BalanceChange } from './parties-json';
+
+export {
+  PublicKey,
+  Field,
+  Bool,
+  VerificationKey,
+  AuthRequired,
+  StringWithHash,
+  Balance,
+  GlobalSlot,
+  CurrencyAmount,
+  StateHash,
+  Fee,
+  BlockTime,
+  UInt32,
+  Signature,
+  TokenId,
+  Sign,
+  SnappProof,
+  Memo,
+};
+
+export { TypeMap };
+
+type Field = string;
+type Bool = boolean;
+type UInt64 = string;
+type UInt32 = string;
+
+type Memo = string;
+type VerificationKey = string;
+type SnappProof = string;
+type Signature = string;
+type PublicKey = string;
+
+type Sign = 'Positive' | 'Negative';
+type AuthRequired = 'Signature' | 'Proof' | 'Either' | 'None' | 'Impossible';
+type StringWithHash = string;
+
+// derived types
+type Balance = UInt64;
+type GlobalSlot = UInt32;
+type CurrencyAmount = UInt64;
+type StateHash = Field;
+type Fee = UInt64;
+type BlockTime = UInt32;
+type TokenId = Field;
+
+// to what types in the js layout are mapped
+type TypeMap = {
+  PublicKey: PublicKey;
```

Comment:
(I think) you can use field punning to reduce the noise here:

```typescript
type TypeMap = {
  PublicKey,
  Field,
  /* ... etc */
}
```

Reply:
no, I checked.. for types it doesn't work